src/polymer/element-descriptor.ts

```diff
@@ -182,33 +182,36 @@ function resolveElement(
 }
 
 function getFlattenedAndResolvedBehaviors(
-    behaviors: string[], document: Document) {
+    behaviors: ScannedBehaviorAssignment[], document: Document) {
   const resolvedBehaviors = new Set<Behavior>();
   _getFlattenedAndResolvedBehaviors(behaviors, document, resolvedBehaviors);
   return resolvedBehaviors;
 }
 
 function _getFlattenedAndResolvedBehaviors(
-    behaviors: string[], document: Document, resolvedBehaviors: Set<Behavior>) {
-  const toLookup = behaviors.slice();
-  for (let behaviorName of toLookup) {
+    behaviors: ScannedBehaviorAssignment[], document: Document,
+    resolvedBehaviors: Set<Behavior>) {
+  for (const behavior of behaviors) {
     // TODO(rictic): once we have a system for passing warnings through, this
     //     could be a mild warning and we could just take the last one in the
     //     array, which should be the most recently defined one.
-    const behavior = document.getOnlyAtId('behavior', behaviorName);
-    if (!behavior) {
-      throw new Error(
-          `In ${document &&
-          document.url}:` +
-              `Unable to resolve behavior \`${behaviorName}\` ` +
-              `Did you import it? Is it annotated with @polymerBehavior?`);
+    const foundBehavior = document.getOnlyAtId('behavior', behavior.name);
```

Comment:
What about pushing the warning onto the document (if the urls match) and continuing? This way we don't stop the analysis

```typescript
if (!foundBehavior) {
  if (document.url === behavior.sourceRange.file) {
    document.warnings.push({/* the warning */});
  }
  continue;
}
```